Added popup DNS list (#1666)
AdguardTeam/AdGuardSDNSFilter: README.md

### Others

* [Additional rules](https://github.com/AdguardTeam/AdGuardSDNSFilter/blob/master/Filters/rules.txt)

## AdGuard DNS filter for popup domains

> The filter is designed to provide a blocking page address in response to popup domain requests. For this purpose the modifiers all, popup and document are replaced by the modifier dnsrewrite with the value of blocking page.

Reviewer: ```suggestion
The filter is designed to show a blocking page when requested a popup domain. For this purpose the modifiers `all`, `popup`, `document` and `document,popup` are replaced by the modifier `dnsrewrite` with the value of blocking page.
```

> The filter is designed to provide a blocking page address in response to popup domain requests. For this purpose the modifiers all, popup and document are replaced by the modifier dnsrewrite with the value of blocking page.

Reviewer 2: This is a bit too technical description.

Let's call it `AdGuard DNS Popup Hosts filter`

Description:

```
DNS-level blocking is a crude method that makes a blocked website appear as if it were malfunctioning from the user's perspective. While this approach may suffice for blocking sub-requests to trackers, it is less effective when a user attempts to access a blocked domain directly.

This filter blocks domains that the users tend to open (more precisely, they are opened against the users' will) and it redirects them to a designated "blocked page," which provides explanations regarding the reasons for blocking.

Although redirecting to a different IP address is not the ideal solution, we are hopeful that the [Structured Errors proposal](https://datatracker.ietf.org/doc/draft-ietf-dnsop-structured-dns-error/) will gain momentum. This would allow us to implement a better solution.
```

> A filter comprising elements from AdGuard filters, EasyList, ABPindo and designed to enhance compatibility with popup blocking at the DNS level. It is crafted to provide the address of a blocking page in response to domain queries.

Reviewer 3: не хватает описания текстом какие именно куски етих фильтров берутся и что делается с правилами. или ето намеренно тут не описано?

> A filter comprising elements from AdGuard filters, EasyList, ABPindo and designed to enhance compatibility with popup blocking at the DNS level. It is crafted to provide the address of a blocking page in response to domain queries.

Reviewer: Если ниже описано из чего состоит, тут наверное не нужно это писать. Просто предназначение фильтра.

> ## AdGuard DNS filter for popup domains structure

Reviewer 2: We should explain here that technically we're only using rules with `$popup`, `$document` and `$all` modifiers as this effectively means that these rules target websites that are opened in the main frame.

* [Additional rules](https://github.com/AdguardTeam/AdGuardSDNSFilter/blob/master/Filters/rules.txt)

## AdGuard DNS Popup Hosts filter

DNS-level blocking is a crude method that makes a blocked website appear as if it were malfunctioning from the user's perspective. While this approach may suffice for blocking sub-requests to trackers, it is less effective when a user attempts to access a blocked domain directly.

This filter blocks domains that the users tend to open (more precisely, they are opened against the users' will) and it redirects them to a designated "blocked page", which provides explanations regarding the reasons for blocking.

Although redirecting to a different IP address is not the ideal solution, we are hopeful that the [Structured Errors proposal](https://datatracker.ietf.org/doc/draft-ietf-dnsop-structured-dns-error/) will gain momentum. This would allow us to implement a better solution.

## AdGuard DNS Popup Hosts filter structure

> The AdGuard DNS Popup Hosts filter includes only rules that contain the modifiers `popup`, `all`, and `domain` from the lists provided. This selection effectively means that these rules are directed at sites that open in the main frame.

Reviewer: `["/^\\|\\|[a-z0-9-.]+\\^(\\$(popup|all|document|document,popup))$/"]`
ещё `document`
